html/semantics/embedded-content/the-iframe-element/iframe_sandbox_allow_top_navigation.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Check that sandboxed iframe can perform navigation on the top frame
+           when allow-top-navigation is set</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+  </head>
+  <body>
+    <iframe sandbox="allow-top-navigation allow-scripts"></iframe>
+    <script>
+      if (opener) {
+        // We're the popup (i.e. a top frame).  Load into the iframe the page
+        // trying to modifying the top frame and transmit the result to our
+        // opener.
+        onmessage = function(e) {
+          opener.postMessage({data: e.data, origin: e.origin}, "*")
+          window.close();
+        }
+        document.querySelector("iframe").src = "support/iframe-that-performs-top-navigation-without-user-gesture-passed.html";
+      } else {
+        // We are the main test page.  Open ourselves as a popup, so that we can
+        // can experiment navigation of the top frame.
+        var t = async_test();
+        onmessage = t.step_func_done(function(e) {
+          assert_equals(e.data.data, "PASS", "Should have the right message");
+        });
+        var popupWin = window.open();
+        popupWin.location.href = location.href;
```

Comment:
Tiny style nit: I'd write this as:

```
async_test(t => {
  window.addEventListener('message', t.step_func_done(e => {
    assert_equals(e.data.data, "PASS");
    e.source.close();
  }));
  window.open(location.href);
}, "Frames with `allow-top-navigation` should be able to navigate the top frame.");
```

Closing the window helps clean things up when you're running the test manually, and I don't think you need the extra `t` or `popupWin` variables.